### PR TITLE
PR: Catch OutOfBoundsDatetime errors from pandas

### LINKED ISSUE #6177: Uncaught error when opening a pandas DataFrame with out of bounds timestamps
## Description

**What steps will reproduce the problem?**

1. Create a Dataframe with list of dictionaries
2. two elements of the dictionary are type of datetime
3. Try open it in spyder via click in variable explorer
4. Throws a lot of exceptions until it crashes

**What is the expected output? What do you see instead?**

**Please provide any additional information below**

  File "C:\Users\pfr\AppData\Local\Continuum\anaconda3\lib\site-packages\spyder\widgets\variableexplorer\dataframeeditor.py", line 260, in get_value
    value = self.df.iat[row, column]
  File "C:\Users\pfr\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\core\indexing.py", line 1786, in __getitem__
    return self.obj.get_value(*key, takeable=self._takeable)
  File "C:\Users\pfr\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\core\frame.py", line 1817, in get_value
    return _maybe_box_datetimelike(series._values[index])
  File "C:\Users\pfr\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\core\common.py", line 172, in _maybe_box_datetimelike
    value = tslib.Timestamp(value)
  File "pandas/_libs/tslib.pyx", line 402, in pandas._libs.tslib.Timestamp.__new__ (pandas\_libs\tslib.c:10051)
  File "pandas/_libs/tslib.pyx", line 1517, in pandas._libs.tslib.convert_to_tsobject (pandas\_libs\tslib.c:28598)
  File "pandas/_libs/tslib.pyx", line 1774, in pandas._libs.tslib._check_dts_bounds (pandas\_libs\tslib.c:32752)
pandas._libs.tslib.OutOfBoundsDatetime: Out of bounds nanosecond timestamp: 9999-01-01 00:00:00

Traceback (most recent call last):
  File "C:\Users\pfr\AppData\Local\Continuum\anaconda3\lib\site-packages\spyder\widge

### PR DESCRIPTION
**Pull request by Prikers**
Fixes #6177

### CODE REVIEW

File: spyder/widgets/variableexplorer/dataframeeditor.py [hunk 1]
```diff
@@ -24,6 +24,7 @@
                             QHeaderView)
 
 from pandas import DataFrame, DatetimeIndex, Series
+from pandas._libs.tslib import OutOfBoundsDatetime
```
Comment (ccordoba12) [inline spyder/widgets/variableexplorer/dataframeeditor.py]: The `pandas._libs` namespace is only available for in Pandas 0.20, but we support much older versions of Pandas.

How could we address that?

Comment (Prikers) [reply spyder/widgets/variableexplorer/dataframeeditor.py]: Oh! I was not aware of that, sorry. What is the oldest version we should support? We can access it from `from pandas.tslib import OutOfBoundsDatetime` in version 0.16. I can check for even older versions. The import will then depend on the pandas version.

Comment (ccordoba12) [reply spyder/widgets/variableexplorer/dataframeeditor.py]: > The import will then depend on the pandas version.
 
Yep, you can change this to

```python
try:
    from pandas._libs.tslib import OutOfBoundsDatetime
except:
    from pandas.tslib import OutOfBoundsDatetime
```

That will work for Pandas 0.13.1+, which is the oldest version we *supposedly* support.

If you think we should update that to a more recent version, feel free to do it in another PR for 3.x :-)

Comment (Prikers) [reply spyder/widgets/variableexplorer/dataframeeditor.py]: Done! Thanks.
For the moment I do not see any strong reason to update the requirement for pandas version but maybe when working on #6008. Is anyone still using 0.13 somewhere? 😄

**Pull request merged**